src/modelBasedFixers/apex-pmd-violations-fixer.ts

```diff
@@ -28,7 +28,7 @@ export class ApexPmdViolationsFixer implements vscode.CodeActionProvider {
         const filteredDiagnostics = context.diagnostics.filter(
             diagnostic => messages.diagnostics.source
               && messages.diagnostics.source.isSource(diagnostic.source)
-              && diagnostic.range.isEqual(range)
+              && range.contains(diagnostic.range)
```

Comment:
When multiple diagnostics are on the same line, the range and diagnostic need not align. The range is calculated from the user interaction and we should just care about that the diagnostic for which we are providing the code fix is within the range of the user selection.